### PR TITLE
Fix - Gamepad support setting "lost focus" behaviour

### PR DESCRIPTION
**Pull request by Greenlamp2**
I noticed when you disable the gamepad support, it keeps the key "RIGHT" pressed.

### CODE REVIEW

**Review by bennybroseph — CHANGES_REQUESTED** [review]
Let's get some comments for this change

File: src/inputs-controller.ts [hunk 1]
```diff
@@ -87,16 +90,41 @@ export class InputsController {
...
+                this.interactions.hasOwnProperty(b) && // if the button is in the interactions dict
+                this.repeatInputDurationJustPassed(b) && // if we hold enough the button
+                this.interactions[b].isPressed // if the button is pressed
+            ) {
+                // if the gamepad support is disable and the source is a gamepad
+                // we don't want to repeat the button
+                // even if we have disabled the gamepad, at the exact moment of the change in the menu
+                // we can be here and have an infinite loop since the code can't know we
+                // have released the key since the gamepad is not there anymore
+                if (!this.gamepadSupport && this.interactions[b].source === 'gamepad') {
```
Comment (bennybroseph) [inline src/inputs-controller.ts:118]: Block comment here

File: src/inputs-controller.ts [hunk 2]
```diff
@@ -251,27 +279,32 @@ export class InputsController {
...
         this.releaseButtonLock(this.buttonLock2);
         for (const b of Utils.getEnumValues(Button)) {
-            if (!this.interactions.hasOwnProperty(b)) return;
-            this.interactions[b].pressTime = null;
-            this.interactions[b].isPressed = false;
+            if (this.interactions.hasOwnProperty(b)) {
+                this.interactions[b].pressTime = null;
+                this.interactions[b].isPressed = false;
+                this.interactions[b].source = null;
+            }
```
Comment (bennybroseph) [inline src/inputs-controller.ts:307]: let's get these functions documented using TSDoc

File: src/inputs-controller.ts [hunk 3]
```diff
@@ -87,16 +90,41 @@ export class InputsController {
...
+        if (value) {
+            this.gamepadSupport = true;
+        } else {
+            this.gamepadSupport = false;
+            // if we disable the gamepad, we want to release every key pressed
+            this.deactivatePressedKey();
+        }
+    }
+
     update(): void {
```
Comment (bennybroseph) [inline src/inputs-controller.ts:105]: let's get these functions documented using TSDoc

File: src/inputs-controller.ts [hunk 4]
```diff
@@ -143,20 +171,20 @@ export class InputsController {
...
             this.events.emit('input_down', {
                 controller_type: 'gamepad',
                 button: buttonDown,
             });
-            this.setLastProcessedMovementTime(buttonDown);
+            this.setLastProcessedMovementTime(buttonDown, 'gamepad');
         }
     }
 
     gamepadButtonUp(pad: Phaser.Input.Gamepad.Gamepad, button: Phaser.Input.Gamepad.Button, value: number): void {
```
Comment (bennybroseph) [inline src/inputs-controller.ts:186]: let's get these functions documented using TSDoc

**Review by bennybroseph — CHANGES_REQUESTED** [review]

**Review by Flashfyre — APPROVED** [review]

**Pull request merged**